# Assignment 1: k-means Clustering
## Group 8
#### Members: Anika Viado, Troy Dela Rosa, Vance Castro, and Ben Thompson

In [1]:
# imports 
import pandas as pd

# **Data Prep**

In [9]:
# load dataset in df
df = pd.read_csv("Country-data.csv")
# print the header 
print(df.columns)

Index(['country', 'child_mort', 'exports', 'health', 'imports', 'income',
       'inflation', 'life_expec', 'total_fer', 'gdpp'],
      dtype='object')


### Summarize the data 

In [8]:
# summarize data
df.describe()

,child_mort,exports,health,imports,income,inflation,life_expec,total_fer,gdpp
count,167.000000,167.000000,167.000000,167.000000,167.000000,167.000000,167.000000,167.000000,167.000000
mean,38.270060,41.108976,6.815689,46.890215,17144.688623,7.781832,70.555689,2.947964,12964.155689
std,40.328931,27.412010,2.746837,24.209589,19278.067698,10.570704,8.893172,1.513848,18328.704809
min,2.600000,0.109000,1.810000,0.065900,609.000000,-4.210000,32.100000,1.150000,231.000000
25%,8.250000,23.800000,4.920000,30.200000,3355.000000,1.810000,65.300000,1.795000,1330.000000
50%,19.300000,35.000000,6.320000,43.300000,9960.000000,5.390000,73.100000,2.410000,4660.000000
75%,62.100000,51.350000,8.600000,58.750000,22800.000000,10.750000,76.800000,3.880000,14050.000000
max,208.000000,200.000000,17.900000,174.000000,125000.000000,104.000000,82.800000,7.490000,105000.000000


The dataset contains data in 167 countries using nine socio-economic and health related factors.

### Column Summary 

**child_mort**:  This column shows the mortality rate of children under 5 years old per 1,000 live births. It has a mean of 38.7 and a median of 19.30, and values ranging from 2.6 and 208. The gap between the mean and medians tells us that some countries have higher child mortatlity than most of the countries in the datat.<br>

**exports**:  This column shows the percentage of the GDP of exported good and services. It has an average of 41.11% and a median of 35.00%, with values ranging from 0.109% to 200%. This tells us that exports contribute differently to each country's ecnonmy.

**health**:  This column shows the percentage of GDP spent on healtcare. It has a mean of 6.82% and median of 6.32% ranging from 1.81% and 17.9%. The small difference between the mean and median suggests that this data is more balanced than the other variables in this dataset.

**imports**:  This column is the percentage GDP of imported goods and services. The values range from 0.0659% to 174%, tell us that countries have very different levels of reliance on imports. The mean of 46.89% and median of 43.30% shows that import levels are similar across amny countries.

**income**:  This is the average per person in a country. With a mean of 17,144.69 and median of 9,960, with a huge range of 609 to 125,000. This shows us that a small portion of countries actually have higher incomes than the majority.

**inflation**:  This shows how fast prices increase in the economy. It has a mean of 7.78%  ranging from -4.21% to 104%. This wide range tells us that some countries experienced different levels of inflation, with some countries having higher values.

**life_expec**:  This shows the number of years a newborn is expected to live. Has a mean of 70.56 years and a median of 73.10 years wiith values ranging from 32.1 to 82.8 years, which shows very different life expectancy across different countries.

**toal_fer**:  This column shows the avreage number of children a woman is expected to have. It has a mean of 2.95 and median of 2.41 ranfinf from 1.15 to 7.49. This shows that the fertility rate varies across the countries as well.

**gdpp**:  GDP per person, or per capita. This column has a mean of 12,964.16 and median of of 4,660 and values ranging from 231 to 105,000. This large difference between the mean and median shows that smaller number of countries have higher GDP per capita than most countries in this dataset. 

The summary statistics of the countries in this dataset shows that they have different ecnonomic, health, and demographic conditions. There are some variables that have large difference between the mean and median, which means there are some countries with much higher or lower values than others. For example, child mortatlity, income, and GDP per capita all show wide ranges, suggesting that some countries ar emuch more developed or have much worse health outcomes than others. Life expectancy, inflation, and fetility rates also cary across countries, which shows clear differences in living standards, economic stability, and etc. These differences are useful because they show meaningful patterns in the dara and amek it possible to group countries with similar characteristics during clustering. These stats provide valuable insights because they hightlihgt how different eacg country is accross the variables in thsi dataset.